### ISSUE TITLE
User API implementation with CRUD operations

### ISSUE THREAD

**Issue by owlsketch** [owlsketch/gallery#61]
**/users**
Read all users
Update all users
I advise against allowing deletion of all users 

**/users/:userid**
Create a user 
Read a user
Update a user 
Delete a user

**Comment by owlsketch** [comment]
I don't think we need the ability to update all users. At least not for now.

**Comment by owlsketch** [comment]
Have written most of this out at 66acc55c571101b6a58d28e0874584e1e6e241a4
However, still need to refactor and implement Delete User

**Comment by owlsketch** [comment]
Done in latest commit! d341ee82752a44c5cb8ede58eeab91c627146205
Will close this once it is merged into dev branch.
